### PR TITLE
Dis 79

### PR DESCRIPTION
**Pull request by goosemanjack**
Adds in debugging hooks and deeper logging for troubleshooting production sites.

DIS-79, DIS-83, DIS-84, DIS-85.

Addition of DebugControl object and methods:

DigiTrust.debugControl.dumpLogs()
DigiTrust.setDebug()

This is initial functionality but should be considered a work in process.

### CODE REVIEW

File: documentation/debugging.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Debugging DigiTrust
+
+A live system may be changed to debug mode in one of two ways: via conf object or via the url hash `dt-debug=true` value.
```
Comment (samtingleff) [inline documentation/debugging.md]: can you provide an example via conf object

File: pages/dt.html [hunk 2]
```diff
@@ -85,6 +98,7 @@
                         }
 
                         if ((!isOptOut) && (helpers.isEmpty(digiTrustUserJSON))) {
+                            log.debug('Create new DigiTrust Id on digitru.st domain');
                             // see if we can write and read cookies from within the iframe
                             DigiTrustCookie.createUserCookiesOnDigitrustDomain();
                             digiTrustUserJSON = DigiTrustCookie.getIdentityCookieJSON(cookieKey);
```
Comment (samtingleff) [inline pages/dt.html]: suggest adding a debug statement here to the effect of `log.debug(digiTrustUserJSON);`. this seems like the most direct place to answer the question of "are cookies working at all".

File: documentation/debugging.md (hunk 1 above)
Comment (goosemanjack) [reply documentation/debugging.md]: This markdown is incorrect - will need to revise. Had a design change to focus on log dumps as outlined here:
[](https://github.com/digi-trust/dt-cdn/wiki/Debugging-DigiTrust)

File: pages/dt.html (hunk 2 above)
Comment (goosemanjack) [reply pages/dt.html]: My only pause on this is exposing the User Id within debug logging. At a highly theoretical level the requestor needs a publisher key to expose Id. Do you have any thoughts on this?

Comment (goosemanjack) [reply pages/dt.html]: Addition of log statement that verifies valid ID without exposing actual ID.

File: documentation/debugging.md (hunk 1 above)
Comment (goosemanjack) [reply documentation/debugging.md]: Improved documents and aligned with actual code.

**Pull request merged**